src/material.js

```diff
@@ -172,25 +172,67 @@ export class Material {
         return new Lattice(this.lattice);
     }
 
+    // returns the derived properties array for a material
+    getderivedProperties() {
+        return this.prop('derivedProperties', []);
+    }
+
+    /**
+     * @summary Returns the a specific derived property of a material, as defined by name
+     * @param {String} name
+     * @returns {Object}
+     */
+    getDerivedPropertyByName(name) {
+        return this.getDerivedProperties().find(x => x.name === name);
+    }
+
     /**
-     * Calculates hash from basis and lattice. Algorithm expects the following:
+     * Calculates the hash for a material.
+     * If `isNonPeriodic` = false
+     * The hash is calculated from the basis and lattice. The algorithm expects the following:
      * - asserts lattice units to be angstrom
      * - asserts basis units to be crystal
      * - asserts basis coordinates and lattice measurements are rounded to hash precision
      * - forms strings for lattice and basis
      * - creates MD5 hash from basisStr + latticeStr + salt
+     *
+     * If `isNonPeriodic` = true
+     * The hash based on the `inchi` string for the material that is returned from getNonPeriodicHashMessage.
+     *
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
     calculateHash(salt = '', isScaled = false) {
-        const message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        let message;
+        if (!this.isNonPeriodic) {
+            message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        } else {
+            message = this.getNonPeriodicHashMessage();
+        }
         return CryptoJS.MD5(message).toString();
     }
 
+    /**
+     * Gets the `inchi` string from the derived properties and returns it as the message to use for
+     * generating the hash when `isNonPeriodic` = true.
+     * @returns String
+     */
+    getNonPeriodicHashMessage() {
+        const inchiString = this.getDerivedProperties() ? this.getDerivedPropertyByName('inchi') : null;
+        if (inchiString) {
+            return inchiString.value;
+        } else {
+            console.error("Non-Periodic hash could not be created. Missing InChI.")
```

Comment:
Should we do the actual error handling - throw an error here? See `ArrayWithIds`

Reply:
I will test this.

Reply:
Still need to test this. In progress now, wanted to get SOF-Stack test pushed